app-agent/src/exec_process.rs

```diff
@@ -17,10 +22,26 @@ use anyhow::Context;
 /// Spawns a child process and waits for it to exit.
 pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result<ExitStatus> {
     // Spawn the process.
-    let mut p = std::process::Command::new(external_command.clone())
-        .args(args)
-        .spawn()
-        .expect("error in child process");
+    let p_result = std::process::Command::new(external_command.clone())
+        .args(args.clone())
+        .spawn();
+
+    let mut p = match p_result {
+        Ok(val) => val,
+        Err(e) => match e.kind() {
+            std::io::ErrorKind::NotFound => {
+                let return_error: String = handle_not_found(external_command, args);
+                panic!("{}", return_error);
+            }
+            std::io::ErrorKind::PermissionDenied => {
+                let return_error: String = handle_permission_denied(external_command);
+                panic!("{}", return_error);
+            }
+            _ => {
+                panic!("Error in child process");
+            }
```

Comment:
The function returns a `Result`, you should use that instead of panics.

Reply:
As in the main you used `expect()` I expected to panic in case of error